streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClientTest.java

```diff
@@ -130,6 +133,17 @@ public void shouldNotAllowNullTopicConfigs() throws Exception {
         verifyCorrectTopicConfigs(streamsKafkaClient, topicConfigWithNoOverrides, Collections.singletonMap("cleanup.policy", "delete"));
     }
 
+    @Test
+    public void metricsShouldBeTaggedWithClientId() {
+        config.put(StreamsConfig.CLIENT_ID_CONFIG, "some_client_id");
+        config.put(StreamsConfig.METRIC_REPORTER_CLASSES_CONFIG, TestMetricsReporter.class.getName());
+        StreamsKafkaClient.create(new StreamsConfig(config));
+        assertFalse(TestMetricsReporter.METRICS.isEmpty());
+        for (KafkaMetric kafkaMetric : TestMetricsReporter.METRICS.values()) {
+            assertEquals("some_client_id", kafkaMetric.metricName().tags().get("client-id"));
```

Comment:
Since just one value to check do we need the `for` loop?

Reply:
There are ~30 metrics registered at this point. I'm just asserting that they are all tagged with the proper client id, rather than how many there are.